src/niamoto/cli/commands/initialize.py

```
@@ -0,0 +1,224 @@
+            os.makedirs(config_dir, exist_ok=True)
+            # By default, our Config class will create config.yml, sources.yml, etc. if absent
+            environment = Environment(config_dir)
+            environment.initialize()
+            print_success("Environment initialized successfully.")
+
+        display_environment_status()
+        display_next_steps()
+
+    except Exception as e:
+        print_error(f"Setup failed: {str(e)}")
+        raise click.Abort()
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)